mlabs-haskell/cardano.nix
# Ogmios package, module and test

fixes #30 #31 #32
- cardano-node 8.8.* and 8.9.* do not build so we're stuck with 8.7.3
- work around cardano-node socket permissions issue: poll and chmod
- work around cardano-node configuration json sharing issue: set from cardano-configurations for all services (found specific version for cardano-node 8.7.3)
- add ogmios binary release ~6.1.0~ 6.2.0 because it fails to build with haskell.nix and 6.0.3 is too old
- add ogmios module from ogmios-nixos
- update ogmios module to work with newer ogmios
- update ogmios module to work with cardano.nix cardano-node
- avoid segfaults, lock down ogmios systemd security
- update module field documentation
- update vmTests (formerly nixosTests) to automatically handle checks, effects, and add apps to run tests
- update vmTests to expose check and effect derivations
- fix run-vm-test argument quotation and shellcheck errors
- add vm test to check ogmios sync progress
- update readme, detail how to run integration test showing cardano-node and ogmios works
